File: src/gestorAplicacion/Cita.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package gestorAplicacion;
import java.time.LocalDateTime;
import java.util.ArrayList;

import gestorAplicacion.personas.*;
/**
 *
 * @author Roger Vera
 */
public class Cita extends Consulta{
	private static final long serialVersionUID = 1L;
    private String motivo;
    private tipoCita tiposCitas;
    int numeroCitas;
    
    public static ArrayList<Cita> listaCitas = new ArrayList<Cita>();
    
    //constructor

    public Cita(int id, Paciente paciente, String motivo, Medico medico, Consultorio consultorio, LocalDateTime fecha, Pago pago, tipoCita tiposCitas) {
        super(id, paciente, medico, consultorio, fecha, pago);
        this.motivo = motivo;
        this.tiposCitas = tiposCitas;
        numeroCitas++;
        listaCitas.add(this);
    }
    public Cita(Paciente paciente, String motivo, Medico medico, Consultorio consultorio, LocalDateTime fecha, Pago pago, tipoCita tiposCitas) {
    	super(paciente, medico, consultorio, fecha, pago);
    	this.motivo = motivo;
    	this.tiposCitas = tiposCitas;
    	listaCitas.add(this);
    }
    
    public Cita(int id, Paciente paciente, String motivo, Medico medico, Consultorio consultorio, LocalDateTime fecha, tipoCita tiposCitas) {
        super(id, paciente, medico, consultorio, fecha, null);
        this.motivo = motivo;
        this.tiposCitas = tiposCitas;
        numeroCitas++;
        listaCitas.add(this);
    }
    public Cita(Paciente paciente, String motivo, Medico medico, Consultorio consultorio, LocalDateTime fecha, tipoCita tiposCitas) {
    	super(paciente, medico, consultorio, fecha, null);
    	this.motivo = motivo;
    	this.tiposCitas = tiposCitas;
    	listaCitas.add(this);
    }
    
    public Cita(int id, Paciente paciente, String motivo, tipoCita tiposCitas) {
    	super(id, paciente, null, null, null, null);
    	this.motivo = motivo;
    	this.tiposCitas = tiposCitas;
    	listaCitas.add(this);
    }
    
    public Cita(Paciente paciente, String motivo, tipoCita tiposCitas) {
    	super(paciente, null, null, null, null);
    	this.motivo = motivo;
    	this.tiposCitas = tiposCitas;
    	listaCitas.add(this);
    }

    //getter & setter
    
    public String getMotivo() {
        return motivo;
    }

    public void setMotivo(String motivo) {
        this.motivo = motivo;
    }

    public tipoCita getTiposCitas() {
        return tiposCitas;
    }

    public void setTiposCitas(tipoCita tiposCitas) {
        this.tiposCitas = tiposCitas;
    }

    public int getNumeroCitas() {
        return numeroCitas;
    }

    public void setNumeroCitas(int numeroCitas) {
        this.numeroCitas = numeroCitas;
    }
    
    //metodos
    
    public static int totalCitasAgendadas(){
        return listaCitas.size();
    }
    
    public int numeroCitasAgendadas(){
        return numeroCitas;
    }
    public String datosCita() {
        return "Paciente: " + getPaciente() + "\nMotivo: " + getMotivo() + 
                "\nMedico: " + getMedico() + "\nConsultorio: " + getConsultorio() +
                "\nFecha: " + getFecha();
    }
    public static int informeCitas(tipoCita solicitado){// se ingresa el tipo de Cita y se busca en la lista cuantas hay
        int tcSolicitada=0;
        for (Cita tc : listaCitas) {
            if(tc.getTiposCitas()==solicitado){
                tcSolicitada++;
            }
        }
        return tcSolicitada;
    }
}

File: Python/src/gestorAplicacion/Cita.py
"""
La clase Cita se utiliza para almacenar los atributos y métodos de las citas agendadas por los pacientes en el sistema.
"""
import random as rd
from gestorAplicacion.Consulta import Consulta
class Cita(Consulta):
    numeroCitas = 0
    listaCitas = []
    def __init__(self, id = int(rd.random() * 10000 + 1), paciente = None, motivo = None, medico = None, 
                 consultorio = None, fecha = None, pago = None, tipoCita = None):
        super().__init__(id, paciente, medico, consultorio, fecha, pago)
        self._motivo = motivo
        self._tipoCita = tipoCita
        Cita.listaCitas.append(self)
    
    # Getters y Setters
    @classmethod
    def getCitas(self):
        return self.listaCitas
    def getMotivo(self):
        return self._motivo
    
    def setMotivo(self, motivo):
        self._motivo = motivo
        
    def getTipoCita(self):
        return self._tipoCita
    
    def setTipoCita(self, tipoCita):
        self._tipoCita = tipoCita

    #metodos
    @classmethod
    def totalCitasAgendadas():
        return len(Cita.listaCitas)
    @classmethod
    def numeroCitasAgendadas():
        return Cita.numeroCitas
    @classmethod
    def datosCita(cls):
        a="Paciente: "+ Consulta.getPaciente()+ "\nMotivo: " + cls.getMotivo + "\nMedico: " + Consulta.getMedico() + "\nConsultorio: " + Consulta.getConsultorio() +"\nFecha: " + Consulta.getFecha()
        return a
    @classmethod
    def informeCitas(cls, solicitado):
        tcSolicitada=0
        for tc in Cita.listaCitas:
            if(tc.getTipoCita == solicitado):
                tcSolicitada +=1
        return tcSolicitada



